app/src/main/java/com/eveningoutpost/dexdrip/Services/DexShareCollectionService.java

```diff
@@ -289,6 +289,11 @@ public void call(EGVRecord[] egvRecords) {
                                         if (egvRecords != null) {
                                             Log.d(TAG, "Made the full round trip, got " + egvRecords.length + " EVG Records");
                                             BgReading.create(egvRecords, addativeSystemTimeOffset, getApplicationContext());
+                                            {
```

Comment:
unfortunately @tzachi-dar this is all happening async so a normal try finally wrapping this would cut off the wakelock probably before a single response was received from the share